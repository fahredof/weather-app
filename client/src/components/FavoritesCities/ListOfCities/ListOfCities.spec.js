import React from 'react';
import ListOfCities from "./ListOfCities";
import crossButton from "./svg/crossButton.svg";

describe(`List of Cities`, () => {
    const mainCity = {
        city: `Ufa`,
        image: `https://openweathermap.org/img/wn/04n@2x.png`,
        temp: `-11`,
        error: ``
    };
    const mainCityError = {
        error: `The server responded with a status of 404`
    };
    const props = {
        cityData: mainCity
        //onClickDelete: () => {}
    };

    describe(`отображение города`, () => {
        const city = shallow(<ListOfCities {...props}/>);
        it(`отображается корректно`, () => expect(shallowToJson(city)).toMatchSnapshot());
    });

    describe('ошибка при получении города', () => {
        const nextProps = {
            ...props,
            cityData: mainCityError
        };
        const city = shallow(<ListOfCities {...nextProps} />);
        it('отображается корректно', () => expect(shallowToJson(city)).toMatchSnapshot());
    });

    describe('загрузка', () => {
        const nextProps = {
            ...props,
            cityData: {}
        };
        const defCity = shallow(<ListOfCities {...nextProps} />);
        it('отображается корректно', () => expect(shallowToJson(defCity)).toMatchSnapshot());
    });

    describe(`Cross button image`, () => {
        const nextProps = {
            ...props,
            cityData: mainCityError
        };
        const button = shallow(<ListOfCities {...nextProps}/>);

        it(`Рендеринг картинки`, () => {
            expect(button.find("img").prop("src")).toEqual(crossButton);
        });

    });
});