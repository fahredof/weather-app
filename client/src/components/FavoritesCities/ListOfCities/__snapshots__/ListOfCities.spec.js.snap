// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DefaultWeather загрузка отображается корректно 1`] = `<Spinner />`;

exports[`DefaultWeather отображение города отображается корректно 1`] = `
<div
  id="list"
>
  <div
    className="city"
  >
    Ufa
  </div>
  <div
    className="temp"
  >
    -11
  </div>
  <div
    className="image"
  >
    <img
      alt=""
      src="https://openweathermap.org/img/wn/04n@2x.png"
    />
  </div>
  <div
    className="sp"
  >
    <Details
      mainCity={
        Object {
          "city": "Ufa",
          "error": "",
          "image": "https://openweathermap.org/img/wn/04n@2x.png",
          "temp": "-11",
        }
      }
    />
  </div>
  <img
    alt=""
    src="crossButton.svg"
  />
</div>
`;

exports[`DefaultWeather ошибка при получении города отображается корректно 1`] = `
<div
  id="list"
>
  <div
    className="wind"
  >
    The server responded with a status of 404
  </div>
  <div
    className="crossButton"
  >
    <img
      alt=""
      src="crossButton.svg"
    />
  </div>
</div>
`;

exports[`List of Cities загрузка отображается корректно 1`] = `<Spinner />`;

exports[`List of Cities отображение города отображается корректно 1`] = `
<div
  id="list"
>
  <div
    className="city"
  >
    Ufa
  </div>
  <div
    className="temp"
  >
    -11
  </div>
  <div
    className="image"
  >
    <img
      alt=""
      src="https://openweathermap.org/img/wn/04n@2x.png"
    />
  </div>
  <div
    className="sp"
  >
    <Details
      mainCity={
        Object {
          "city": "Ufa",
          "error": "",
          "image": "https://openweathermap.org/img/wn/04n@2x.png",
          "temp": "-11",
        }
      }
    />
  </div>
  <img
    alt=""
    src="crossButton.svg"
  />
</div>
`;

exports[`List of Cities ошибка при получении города отображается корректно 1`] = `
<div
  id="list"
>
  <div
    className="wind"
  >
    The server responded with a status of 404
  </div>
  <div
    className="crossButton"
  >
    <img
      alt=""
      src="crossButton.svg"
    />
  </div>
</div>
`;
